dbos/_core.py

```diff
@@ -783,21 +784,22 @@ def decorate_transaction(
 ) -> Callable[[F], F]:
     def decorator(func: F) -> F:
 
-        transactionName = func.__qualname__
+        transaction_name = func.__qualname__
 
         def invoke_tx(*args: Any, **kwargs: Any) -> Any:
             if dbosreg.dbos is None:
                 raise DBOSException(
                     f"Function {func.__name__} invoked before DBOS initialized"
                 )
 
+            dbos = dbosreg.dbos
             ctx = assert_current_dbos_context()
-            if dbosreg.is_workflow_cancelled(ctx.workflow_id):
+            status = dbos._sys_db.get_workflow_status(ctx.workflow_id)
+            if status and status["status"] == WorkflowStatusString.CANCELLED.value:
```

Comment:
IIUC we can't abort in the middle of the retries because the retry loop never checks the system database... correct?

Reply:
Yes this is outside of the transaction retry loop.